### PR TITLE
Appng 120 - Include better support for mock development

### PR DESCRIPTION
**Pull request by jramcast**
Used angular-mocks to provide mocked backends through `$httpBackend`.

This service is injected if a configuration value is enabled (`REST_CONFIG.MockBackend`)

### CODE REVIEW

**Comment by jramcast** [comment]
Updated branch to remove conflicts

File: demo/partials/content.html [hunk 1]
```diff
@@ -143,7 +143,7 @@
         <h2>REST</h2>
         <h3>Load Books with rest directive</h3>
 
-        <div rest rest-path="data/books.json" rest-name="mybooks"></div>
+        <div rest rest-path="books.json" rest-name="mybooks"></div>
```
Comment (aecz) [inline demo/partials/content.html]: Request suffix is set to .json so extension should not be needed.

File: gruntfile.js [hunk 2]
```diff
@@ -345,7 +345,6 @@ module.exports = function (grunt) {
                     port: 9000,
                     middleware: function (connect) {
                         return [
-                            delayApiCalls,
```
Comment (aecz) [inline gruntfile.js]: Delay is used in REST demo to allow the user to see the loading message. Should be kept.

Comment (jramcast) [inline gruntfile.js]: The problem with this is that it makes e2e tests very slow... Maybe we can just trigger the REST call when clicking a button

**Comment by jramcast** [comment]
Ready to be merged

**Pull request merged**